man/fi_gni.7.md

```diff
@@ -56,6 +56,16 @@ libfabric API:
 : For both control and data progress, the GNI provider supports both
   *FI_PROGRESS_AUTO* and *FI_PROGRESS_MANUAL*, with a default set to
   *FI_PROGRESS_AUTO*.
+  Note that for data progress, progression is only performed
+  when data transfers use the rendezvous protocol.
+
+*Wait Objects*
+: The GNI provider specifically supports wait object types *FI_WAIT_UNSPEC*,
+  and *FI_WAIT_SET*. A wait object must be used when calling fi_cntr_wait,
+  fi_cq_sread/from, fi_eq_sread/from, fi_wait.
+  The GNI provider spawns an internal wait progress thread that is woken when
+  calls into the providers wait system occur, it is only active during calls
```

Comment:
"calls into the providers wait system occur" -> "clients utilize the wait system (e.g., calling fi_wait)."

End sentence there.  The second (run-on) clause makes things more confusing.